assets/schemas/manifest.json

```diff
@@ -696,121 +1319,284 @@
+          "name": {
+            "type": "string"
           },
-          "minLength": {
-            "type": "integer",
-            "format": "int64",
-            "default": 0
+          "description": {
+            "type": "string"
           },
-          "maxLength": {
-            "type": "integer",
-            "format": "int64",
-            "default": 0
+          "tcfId": {
+            "type": "integer"
           },
-          "pattern": {
+          "tcfType": {
             "type": "string",
-            "default": ""
+            "enum": [
+              "purpose",
+              "specialPurpose",
+              "feature",
+              "specialFeature"
+            ]
           },
-          "placeholder": {
-            "type": "string",
-            "description": "The placeholder text that will appear when this field is empty",
-            "default": ""
+          "editable": {
+            "type": "boolean",
+            "default": false
           },
-          "default": {
-            "type": ["string", "null"],
-            "description": "Default value will be the value for this field, before user interaction. Having a default value will override the placeholder text"
+          "legalBasisRestriction": {
+            "type": "string"
           },
-          "spellcheck": {
-            "type": "boolean",
-            "description": "This setting will enable spell check on the field.",
-            "default": true
+          "displayName": {
+            "type": "string"
           },
-          "multiple": {
-            "type": "boolean",
-            "description": "If true, multiple values can be added in this field.\n The values will appear as an array in the API and an “Add Another” button will be visible on the field allowing the creation of additional fields for this component",
-            "default": false
+          "displayDescription": {
+            "type": "string"
           },
-          "data": {
+          "processingPurpose": {
+            "type": "string"
+          },
+          "legalBasis": {
             "type": "object",
-            "properties": {
-              "oneOf": {
-                "values": {
-                  "type": ["array", "null"],
-                  "description": "These are the values that will be selected on this field",
-                  "items": {
-                    "type": "object",
-                    "required": [
-                      "label",
-                      "value"
-                    ],
-                    "additionalProperties": false,
-                    "properties": {
-                      "value": {
-                        "type": "string"
-                      },
-                      "label": {
-                        "type": "string"
-                      }
-                    }
-                  }
+            "additionalProperties": {
+              "type": "string"
+            }
```

Comment:
does this map from policy scope to legal basis? what is stored in object?

Reply:
`regulation` -> `legalBasis`

Reply:
so this is saying given a policy scope with regulation x select legal basis y for the purpose? any thoughts about policy scopes with conflicting regulations?